front/src/modules/timesStops/hooks/useTimesStopsTableData.ts

```diff
@@ -0,0 +1,254 @@
+
+type BuildTableRowParams = {
+  id: string;
+  name?: string;
+  secondaryCode?: string;
+  trackName?: string;
+  startDate: Date;
+  schedule?: ScheduleItem;
+  computedArrivalMs?: number;
+};
+
+/** Row data without the index field (added after all rows are built) */
+type TimesStopsRowWithoutIndex = Omit<TimesStopsRowNew, 'index'>;
+
+const buildTableRow = ({
+  id,
+  name,
+  secondaryCode,
+  trackName,
+  startDate,
+  schedule,
+  computedArrivalMs,
+}: BuildTableRowParams): TimesStopsRowWithoutIndex => {
+  // schedule.arrival is ISO 8601 duration
+  const requestedArrival = schedule?.arrival
+    ? new Date(startDate.getTime() + Duration.parse(schedule.arrival).ms)
+    : null;
+
+  // computedArrivalMs is offset in ms from startDate
+  const computedArrival =
+    computedArrivalMs !== undefined ? new Date(startDate.getTime() + computedArrivalMs) : null;
+
+  // schedule.stop_for is ISO 8601 duration
+  const stopDuration = schedule?.stop_for ? Duration.parse(schedule.stop_for) : null;
+
+  // requestedDeparture = requestedArrival + stopDuration
+  const requestedDeparture =
+    requestedArrival && stopDuration !== null
+      ? new Date(requestedArrival.getTime() + stopDuration.ms)
+      : null;
+
+  // computedDeparture = computedArrival + stopDuration
+  const computedDeparture =
+    computedArrival && stopDuration !== null
+      ? new Date(computedArrival.getTime() + stopDuration.ms)
+      : null;
+
+  return {
+    id,
+    name: name ?? '',
+    secondaryCode: secondaryCode ?? '',
+    track: trackName ?? '',
+    requestedArrival,
+    computedArrival,
+    stopDuration,
+    requestedDeparture,
+    computedDeparture,
+  };
+};
+
+/**
+ * Hook to build TableRow[] for the new Times & Stops table (TimesStopsTableNew).
+ * This is separate from useOutputTableData to keep the legacy and new logic independent.
+ * Once the new table is fully validated, useOutputTableData can be removed.
+ */
+const useTimesStopsTableData = ({
+  infraId,
+  isValid,
+  selectedTrain,
+  simulatedTrain,
+  simulatedPathItemTimes,
+  operationalPointsOnPath,
+  skip = false,
+}: {
+  infraId: number;
+  isValid: boolean;
+  selectedTrain: Train;
+  simulatedTrain?: SimulationResponseSuccess['final_output'];
+  simulatedPathItemTimes?: Extract<SimulationSummary, { isValid: true }>['pathItemTimes'];
+  operationalPointsOnPath?: PathPropertiesFormatted['operationalPoints'];
+  /** Skip rows computation. TODO: remove when the new times stops table is fully developed */
+  skip?: boolean;
+}): TimesStopsRowNew[] => {
+  const { t } = useTranslation('operational-studies');
+  const { getTrackSectionsByIds } = useScenarioContext();
+  const displayOnlyPathSteps = useSelector(getDisplayOnlyPathSteps);
+
+  const pathStepOps = usePathOps(infraId, selectedTrain?.path);
+
+  const trackIds = useMemo(() => {
+    const pathSteps = selectedTrain?.path || [];
+    const trackIdsInPathSteps: string[] = [];
+    for (const { location } of pathSteps) {
```

Comment:
`selectedTrain` is always defined, I think you can drop the intermediate const
```suggestion
    const trackIdsInPathSteps: string[] = [];
    for (const { location } of selectedTrain.path) {
```

Reply:
fixed